src/JSInterop/Microsoft.JSInterop/src/Infrastructure/TaskGenericsUtil.cs

```diff
@@ -77,37 +56,4 @@ private sealed class VoidTaskResultGetter : ITaskResultGetter
             return null;
         }
     }
-
-    private sealed class TcsResultSetter<T> : ITcsResultSetter
-    {
-        public Type ResultType => typeof(T);
-
-        public void SetResult(object tcs, object? result)
-        {
-            var typedTcs = (TaskCompletionSource<T>)tcs;
-
-            // If necessary, attempt a cast
-            var typedResult = result is T resultT
-                ? resultT
-                : (T)Convert.ChangeType(result, typeof(T), CultureInfo.InvariantCulture)!;
-
-            typedTcs.SetResult(typedResult!);
-        }
-
-        public void SetException(object tcs, Exception exception)
-        {
-            var typedTcs = (TaskCompletionSource<T>)tcs;
-            typedTcs.SetException(exception);
-        }
-    }
-
-    private static ITcsResultSetter CreateResultSetter(object taskCompletionSource)
-    {
-        return _cachedResultSetters.GetOrAdd(taskCompletionSource.GetType(), tcsType =>
-        {
-            var resultType = tcsType.GetGenericArguments()[0];
-            return (ITcsResultSetter)Activator.CreateInstance(
-                typeof(TcsResultSetter<>).MakeGenericType(resultType))!;
-        });
-    }
```

Comment:
We were using this reflection code in JS interop, but I replaced it with a different abstraction that doesn't require reflection (`JSInteropTask`)